CalPatRec/src/CalHelixFinderAlg.cc

```diff
@@ -139,48 +139,50 @@ namespace mu2e {
-    _maxPanelToHelixDPhi(pset.get<float>         ("maxPanelToHelixDPhi"    )), // 75 degrees
-    _distPatRec         (pset.get<float>         ("distPatRec"             )),
-    _minDeltaNShPatRec  (pset.get<float>         ("minDeltaNShPatRec"      )),
-    _mindist            (pset.get<float>         ("mindist"                )),
-    _pmin               (pset.get<float>         ("minP"                   )),
-    _pmax               (pset.get<float>         ("maxP"                   )),
-    _tdmin              (pset.get<float>         ("minAbsTanDip"           )),
-    _tdmax              (pset.get<float>         ("maxAbsTanDip"           )),
-    _xyweights          (pset.get<bool>          ("xyWeights"              )),
-    _zweights           (pset.get<bool>          ("zWeights"               )),
-    _filter             (pset.get<bool>          ("filter"                 )),
-    _plotall            (pset.get<bool>          ("plotall"                )),
-    _usetarget          (pset.get<bool>          ("usetarget"              )),
-    _maxZTripletSearch  (pset.get<float>         ("maxZTripletSearch"      )),
-    _bz                 (0.0),
-    _nHitsMaxPerPanel   (pset.get<int>           ("nHitsMaxPerPanel"       )),
-    _hitChi2Max         (pset.get<float>        ("hitChi2Max"             )),
-    _chi2xyMax          (pset.get<float>        ("chi2xyMax"              )),
-    _chi2zphiMax        (pset.get<float>        ("chi2zphiMax"            )),
-    _chi2hel3DMax       (pset.get<float>        ("chi2hel3DMax"           )),
-    _dfdzErr            (pset.get<float>        ("dfdzErr"                )){
-
-    float minarea(pset.get<float>("minArea"));
+
+  CalHelixFinderAlg::CalHelixFinderAlg(const Config& config) :
+    _diag(config.diag()),
+    _debug(config.debug()),
+    _debug2(config.debug2()),
+    _hsel(config.hsel()),
+    _bkgsel(config.bkgsel()),
+    _maxHitEnergy(config.maxHitEnergy()),
+    _minNHits(config.minNHits()),
+    _absMpDfDz(config.absMpDfDz()),
+    _initDfDz(config.initDfDz()),
+    _dzOverHelPitchCut(config.dzOverHelPitchCut()),
+    _maxDfDz(config.maxDfDz()),
+    _minDfDz(config.minDfDz()),
+    _sigmaPhi(config.sigmaPhi()),
+    _weightXY(config.weightXY()),
+    _targetcon(config.targetcon()),
+    _weightZPhi(config.weightZPhi()),
+    _weight3D(config.weight3D()),
+    _maxXDPhi(config.maxXDPhi()),
+    _maxPanelToHelixDPhi(config.maxPanelToHelixDPhi()),
+    _distPatRec(config.distPatRec()),
+    _minDeltaNShPatRec(config.minDeltaNShPatRec()),
+    _mindist(config.mindist()),
+    _pmin(config.pmin()),
+    _pmax(config.pmax()),
+    _tdmin(config.tdmin()),
+    _tdmax(config.tdmax()),
+    _xyweights(config.xyweights()),
+    _zweights(config.zweights()),
+    _filter(config.filter()),
+    _plotall(config.plotall()),
+    _usetarget(config.usetarget()),
+    _maxZTripletSearch(config.maxZTripletSearch()),
+    _bz(config.bz()),
```

Comment:
this needs to be removed

Reply:
I'm referring to the _bz(....). That parameter is set in the BeginRun, so it doesn't need to be initialized here.